### DISCUSSION TITLE
So how am i supposed to use flame with BloC or provider packages ?

### DISCUSSION THREAD

**Discussion by FotisDovas** [flame-engine/flame#1435] [Q&A] [answered]
Hello there!
I am planning on using the flame engine for my project this year, and the concept seems doable. Only problem is i want to implement some sensor-datastream reads (a package that works for me outside flame is motion_sensors), and i cant find a way to read them inside the engine, until now. The use of a provider package will be necessary for other things too, i am sure. Can someone help me ?

**Comment by erickzanardo** [comment, marked as answer]
Hey, this may help you: https://verygood.ventures/blog/flutter-games-with-bloc-and-flame

**Reply by FotisDovas** [reply]
UUhhh real nice! I will check, right away! Since you are here, you think there is something similar to streamBuilder for flame components ? (i will be checking the article too of course)

**Reply by erickzanardo** [reply]
I don't think we do, at least not anything from the core team. But if you could give an usage example, we can discuss a feature request for it

**Reply by FotisDovas** [reply]
Yeah sure. The sensor data (mostly from orientation tilts and rolls) will be streamed as to create Offsets (or Vector2) for the position of the main 'character'. the viewport and camera is locked and static. This should work exactly like a pan gesture control, but instead of gesture details, we will have some clamped orientation data (in radians or degrees()), whose deltas will be giving as a movement orientation vector and maybe also a speed. It is fairly simple as a concept. I think i would implement it by reading the current value of the sensor stream, per renderer frame, since that sensor stream can have varying sample rate (motion_sensors can easily reach 60fps reads on my 3 year old android phone).

**Reply by erickzanardo** [reply]
I wonder if we need a something like a stream builder though. A flame game is already a stateful entity, so you could just listen to stream on the game onLoad, and close the stream subscription on the game remove.

We could ofc maybe provide some shortcuts and helpers for that, but I do believe this should be straight forward even doing manually.

**Reply by FotisDovas** [reply]
Oh great! A listener was all i needed to be honest (+ a provider integration). Since you tell me i can do it onLoad (similar to initState i guess) then it is all set! Thank you <3

P.S can you maybe point me to an example of a stream subscription in flame, since i couldn't find it in the docs yet

**Reply by erickzanardo** [reply]
I honestly don't think we have anything, but maybe the source code of `flame_bloc` can point you the right direction?

https://github.com/flame-engine/flame/blob/main/packages/flame_bloc/lib/src/flame_bloc_game.dart#L60

**Reply by FotisDovas** [reply]
All right! Thanks again i am hyped
